cmake/target/version/generate_version_info.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+"""
+# ===============================================================================
+# NAME: generate_version_info.py
+#
+# DESCRIPTION:  Creates a version.hpp and version.json file with version informations
+#
+# USAGE: ./generate_version_info.py /path/to/outdir/
+#
+# AUTHOR: sfregoso
+# EMAIL:  sfregoso@jpl.nasa.gov
+# DATE CREATED  : Oct. 15, 2021
+#
+# Copyright 2021, California Institute of Technology.
+# ALL RIGHTS RESERVED. U.S. Government Sponsorship acknowledged.
+# ===============================================================================
+"""
+import argparse
+import os
+import sys
+import json
+from pathlib import Path
+
+from fprime_ac.utils.version import (
+    FALLBACK_VERSION,
+    get_fprime_version,
+    get_project_version,
+    get_library_versions,
+)
+
+
+def create_version_file_hpp(output_dir, framework_version, project_version):
+    """
+    Create the version file using the provided name and path.
+    """
+    version_hpp = Path(output_dir) / "version.hpp.tmp"
+    # Open file for writing
+    with open(version_hpp, "w") as fid:
+        fid.write("/*\n")
+        fid.write(
+            f"    This file has been autogenerated using [{os.path.basename(__file__)}].\n"
+        )
+        fid.write("    This file may be overwritten.\n")
+        fid.write("*/\n")
+        fid.write("#ifndef _VERSION_HPP_\n")
+        fid.write("#define _VERSION_HPP_\n")
+        fid.write("\n")
+        fid.write(f'static const char* FRAMEWORK_VERSION = "{framework_version}";\n')
+        fid.write(f'static const char* PROJECT_VERSION = "{project_version}";\n')
+        if len(get_library_versions()) > 0:
+            fid.write("static const char* LIBRARY_VERSIONS[] = {\n")
+            for lib, version in get_library_versions().items():
+                fid.write(f'    "{lib}@{version}",\n')
+            fid.write("};\n")
```

Comment:
I am not very happy with this but I could not find a way to appease the compiler when LIBRARY_VERSIONS was unused and empty... let me know if you can think of something better to do

Reply:
Here's the error I would get
```
versions/version.hpp:10:20: error: zero-size array ‘LIBRARY_VERSIONS’
   10 | static const char* LIBRARY_VERSIONS[] = {
      |                    ^~~~~~~~~~~~~~~~
versions/version.hpp:10:20: error: ‘LIBRARY_VERSIONS’ defined but not used [-Werror=unused-variable]
```